internal/gatewayapi/runner/runner.go

```diff
@@ -61,15 +61,17 @@ func (r *Runner) subscribeAndTranslate(ctx context.Context) {
 		gatewayClasses := r.ProviderResources.GetGatewayClasses()
 		// Fetch the first gateway class since there should be only 1
 		// gateway class linked to this controller
-		t := &gatewayapi.Translator{
-			GatewayClassName: v1beta1.ObjectName(gatewayClasses[0].GetName()),
+		if gatewayClasses == nil || gatewayClasses[0] != nil {
```

Comment:
above line will panic on gateway class deletions
consider this change
if gatewayClasses == nil 
// no need to run translate, publish empty IRs
r.XdsIR.Store(r.Name(), nil)
r.InfraIR.Store(r.Name(), nil)
} else {
 Translate .....
 Store results......
}